test/unit/deleted_object_test.rb

```diff
@@ -16,4 +16,74 @@ class DeletedObjectTest < ActiveSupport::TestCase
     assert_same_elements metrics.map(&:id),   DeletedObject.metrics.pluck(:object_id)
     assert_same_elements contracts.map(&:id), DeletedObject.contracts.pluck(:object_id)
   end
+
+  test '.service_owner' do
+    metric = FactoryBot.create(:metric)
```

Comment:
I will do some refactoring in the tests of this class bcz there is a lot of repetition right now 😄